src/RandN/Extensions/DistributionExtensions.cs

```diff
@@ -0,0 +1,64 @@
+using System;
+using System.Diagnostics.CodeAnalysis;
+
+namespace RandN.Extensions
+{
+    /// <summary>
+    /// Contains extension methods for the <see cref="IDistribution{TResult}"/> interface.
+    /// </summary>
+    public static class DistributionExtensions
+    {
+        /// <summary>
+        /// Transforms a distribution by mapping its values using the selector provided.
+        /// This method implements the "map" operator from functional programming principles.
+        /// </summary>
+        /// <example>
+        /// This method enables use of LINQ query syntax over <see cref="IDistribution{TResult}"/>. For example:
+        /// <code>
+        /// IDistribution&lt;int&gt; evenNumbersDistribution =
+        ///     from i in Uniform.NewInclusive(0, 5)
+        ///     select i * 2;
+        /// </code>
+        /// </example>
+        /// <typeparam name="TSource">The generic type of the input distribution.</typeparam>
+        /// <typeparam name="TResult">The generic type of the output distribution.</typeparam>
+        /// <param name="distribution">The distribution to be transformed.</param>
+        /// <param name="selector">The projection to be applied to values from the distribution.</param>
+        public static IDistribution<TResult> Select<TSource, TResult>(
+            this IDistribution<TSource> distribution,
+            Func<TSource, TResult> selector) =>
+            new SelectDistribution<TSource, TResult>(distribution, selector);
+
+        private class SelectDistribution<TSource, TResult> : IDistribution<TResult>
```

Comment:
I try to make every class `sealed` unless I explicitly need inheritance for some reason.
(Also testing out GitHub suggestions?)
```suggestion
        private sealed class SelectDistribution<TSource, TResult> : IDistribution<TResult>
```

Reply:
Good call - I've applied the suggestion.